lib/assets/javascripts/src/local-time/element_observations.coffee

```diff
@@ -0,0 +1,52 @@
+import LocalTime from "./local_time"
+
+{elementMatchesSelector} = LocalTime
+
+class LocalTime.ElementObservations
+  OBSERVABLE_ATTRIBUTES = [ "datetime", "data-local", "data-format", "data-format24" ]
+
+  constructor: (@selector, @callback) ->
+    @elements = new Map()
+
+  include: (element) =>
+    unless @elements.get(element)
+      observer = @observe(element)
+      @register(element, observer)
+
+  disregard: (element) =>
+    if observer = @elements.get(element)?.observer
+      observer.disconnect()
+      @elements.delete(element)
+
+  observe: (element) =>
+    observer = new MutationObserver(@processMutations)
+    observer.observe(element, characterData: true, subtree: true, attributes: true, attributeFilter: OBSERVABLE_ATTRIBUTES)
+    observer
+
+  register: (element, observer) =>
+    @elements.set(element, { observer: observer, updates: -1 })
+    @incrementUpdates(element)
+
+  processMutations: (mutations) =>
+    for mutation in mutations
+      target = mutation.target
+
+      element = if target.nodeType is Node.TEXT_NODE
+        target.parentNode
+      else
+        target
+
+      if elementMatchesSelector(element, @selector)
+        @processLingeringElement(element)
+        break
+
+  processLingeringElement: (element) =>
+    @incrementUpdates(element)
+    @callback(element)
+
+  incrementUpdates: (element) =>
+    @elements.get(element).updates++
+    element.setAttribute("data-updates", @elements.get(element).updates)
```

Comment:
Only done for testing purposes.

We mutate the element as a response to mutations, so this is prone to infinite loops if we're not careful. I think asserting the number of updates is a good protection against this.